src/protagonist/DLCS.Repository/Assets/ThumbnailCalculator.cs

```diff
@@ -25,13 +25,54 @@ public static SizeCandidate GetCandidate(List<Size> sizes, ImageRequest imageReq
 
     private static SizeCandidate GetLongestEdge(List<Size> sizes, ImageRequest imageRequest)
     {
-        if (imageRequest.Size.Width > 0 && imageRequest.Size.Height > 0)
+        var requestWidth = imageRequest.Size.Width ?? 0;
+        var requestHeight = imageRequest.Size.Height ?? 0;
+        
+        if (requestWidth > 0 && requestHeight > 0)
         {
-            // We don't actually need to check imageRequest.Size.Confined (!w,h) because same logic applies...
-            var max = Math.Max(imageRequest.Size.Width ?? 0, imageRequest.Size.Height ?? 0);
-            return sizes.Select(s => s.MaxDimension).Contains(max)
-                ? new SizeCandidate(max)
-                : new SizeCandidate();
+            // get the longest dimension of the requested size
+            var max = Math.Max(requestWidth, requestHeight);
+            var foundExactSize = sizes.Exists(s => 
+                s.Width == requestWidth &&
+                s.Height == requestHeight);
+
+            // We found a size that matches the request exactly, so we'll go with that
+            if (foundExactSize)
+            {
+                return new SizeCandidate(max);
+            }
+            
+            // If the image is confined, are there any sizes that fit?
+            if (imageRequest.Size.Confined)
+            {
+                // Pick the first thumbnail size as a reference for shape
+                var shape = sizes.First().GetShape();
+
+                // If this is a landscape image, max should match its width
+                if (shape == ImageShape.Landscape 
+                    && sizes.Exists(s => s.Width == max && requestHeight >= s.Height))

```

Comment:
Removing `&& requestHeight >= s.Height` from this query doesn't fail any tests - it should to avoid it accidentally being removed in the future (same with `&& requestWidth >= s.Width` below)

Reply:
Tests for ensuring that the correct shortest edge is being picked have now been added in https://github.com/dlcs/protagonist/pull/699/commits/6d796dca1c816b35d4f351a044f1c04da903cba6